lib/parser/assignment-parser.ts

```diff
@@ -26,19 +26,62 @@ export class AssignmentParser extends ParserBase {
 
 
     this.consumeToken();
-    assignment.guarded = this.maybe('guarded') !== false;
-    if (this.maybe('transport') === false) {
+    const forceToken = this.maybe('force');
+    if (forceToken) { // 10.5.1
+      if (mode !== 'sequential') {
+        this.state.messages.push({
+          message: 'force only allowed in sequential statement',
+          range: forceToken.range
+        });
+      }
+      this.maybe(['in', 'out']);
+      const [rightHandSide] = this.advanceParenthesisAware([';'], true, true);
+      const expressionParser = new ExpressionParser(this.state, assignment, rightHandSide);
+      assignment.references.push(...expressionParser.parse());
+      assignment.range = assignment.range.copyWithNewEnd(this.state.pos.i);
+      this.debug('parse end');
+      return assignment;
+    }
+    const releaseToken = this.maybe('release');
+    if (releaseToken) { // 10.5.1
+      if (mode !== 'sequential') {
+        this.state.messages.push({
+          message: 'force only allowed in sequential statement',
+          range: releaseToken.range
+        });
+      }
+      this.maybe(['in', 'out']);
+      this.expect(';');
+      return assignment;
+    }
+    const guardedToken = this.maybe('guarded');
+    assignment.guarded = guardedToken !== false;
+    if (guardedToken && mode !== 'concurrent') {
+      this.state.messages.push({
+        message: 'guarded only allowed in sequential statement',
+        range: guardedToken.range
+      });
+    }
+    if (this.maybe('transport')) { // 10.5.2.1 delay_mechanism
+    } else {// 10.5.2.1 delay_mechanism inertial
```

Comment:
invert the if condition (didn't we hava a linter rule against empty blocks?) or add todo comment

Reply:
Yeah, the comment tricks the ESLint rule ;)
This is a bit annoying because the if actually has a side effect. (Increases the parser position. For this I do not really like it inverted. Maybe i do it like if (token == 'transport') {consume token} else {... ? What do you think?
I do not like hiding the consumption of transport behind the inverted condition